src/py/_pyodide/console.py

```diff
@@ -362,10 +372,7 @@ def formatsyntaxerror(self, e: Exception) -> str:
         sys.last_type = type(e)
         sys.last_value = e
         sys.last_traceback = None
-        try:
-            return "".join(traceback.format_exception_only(type(e), e))
-        finally:
-            e = None  # type: ignore
```

Comment:
Ah I guess this isn't needed since e wasn't thrown down the stack from this function, so e's traceback won't contain a reference to this scope hence no reference loop to break.

Reply:
If it does cause a problem, I think you want `del e` rather than `e = None`.

Reply:
I copied this from the Python source code. Which doesn't mean that it is the right way to do things of course. It seems that in the CPython source code for v3.10.2 there are 32 instances of `finally: some_var = None` and 36 instances of `finally: del some_var`.

Reply:
Thanks for the explanation!